gluster-file-system/azuredeploy.json

```diff
@@ -0,0 +1,318 @@
+            "type": "string",
+            "metadata": {
+                "description": "virtual network the nodes will be created in"
+            }
+        },
+        "adminUsername": {
+            "type": "string",
+            "metadata": {
+                "description": "ssh user name"
+            }
+        },
+        "adminPassword": {
+            "type": "securestring",
+            "metadata": {
+                "description": "ssh password"
+            }
+        },
+        "imagePublisher": {
+            "type": "string",
+            "defaultValue": "OpenLogic",
+            "allowedValues": [ 
+                "OpenLogic"
+            ],
+            "metadata": {
+                "description": "the publisher name of the OS image"
+            }
+        },
+        "imageOffer": {
+            "type": "string",
+            "defaultValue": "CentOS",
+            "allowedValues": [ 
+                "CentOS"
+            ],
+            "metadata": {
+                "description": "the offer name (OS family) of the OS image"
+            }
+        },
+        "imageSku": {
+            "type": "string",
+            "defaultValue": "6.5",
+            "allowedValues": [ 
+                "6.5"
+            ],
+            "metadata": {
+                "description": "the SKU name of the image offering typically major OS version"
+            }
+        },
+        "vmSize": {
+            "type": "string",
+            "defaultValue": "Standard_A1",
+            "allowedValues": [
+                "Standard_A1",
+                "Standard_A2",
+                "Standard_A3",
+                "Standard_A4"
+            ],
+            "metadata": {
+                "description": "VM size for the nodes"
+            }
+        },
+        "vmNamePrefix": {
+            "type": "string",
+            "metadata": {
+                "description": "VM name prefix, a number will be appended for each node"
+            }
+        },
+        "vmIPPrefix": {
+            "type": "string",
+            "defaultValue": "10.0.0.1",
+            "metadata": {
+                "description": "VM IP prefix, a number will be appened for each node"
+            }
+        },
+        "vnetAddressPrefix": {
+            "type": "string",
+            "defaultValue": "10.0.0.0/16",
+            "metadata": {
+                "description": "virtual network CIDR in which the cluster will be created"
+            }
+        },
+        "gfsSubnetName": {
+            "type": "string",
+            "defaultValue": "Subnet-1",
+            "metadata": {
+                "description": "name of the subnet for the Gluster nodes"
+            }
+        },
+        "gfsSubnetPrefix": {
+            "type": "string",
+            "defaultValue": "10.0.0.0/24",
+            "metadata": {
+                "description": "CIDR of the subnet for the Gluster nodes"
+            }
+        },
+        "customScriptFilePath": {
```

Comment:
My bash script doesn't necessarily configure the linux VM exactly the user wanted.  User should be able to copy the bash script to another location, and customize it. If I make it a variable, they can't do that any more.